services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -7254,13 +7254,16 @@ public Optional<PushStatusStoreRecordDeleter> getPushStatusStoreRecordDeleter()
   }
 
   /**
-   * @see Admin#getEmergencySourceRegion()
+   * @see Admin#getEmergencySourceRegion(String)
    */
   @Override
-  public Optional<String> getEmergencySourceRegion() {
-    return multiClusterConfigs.getEmergencySourceRegion().equals("")
-        ? Optional.empty()
-        : Optional.of(multiClusterConfigs.getEmergencySourceRegion());
+  public Optional<String> getEmergencySourceRegion(@Nonnull String clusterName) {
+    String emergencySourceRegion = multiClusterConfigs.getEmergencySourceRegion(clusterName);
+    if (StringUtils.isNotEmpty(emergencySourceRegion)) {
```

Comment:
nit: can use one line for this
`return StringUtils.isNotEmpty(emergencySourceRegion) ? Optional.of(emergencySourceRegion): Optional.empty()` or even
`Optional.ofNullable(emergencySourceRegion).filter(s -> StringUtils.isNotEmpty(s));`

Reply:
Need to merge . Dropping this for now. Will have more opportunity to improve later